framework/src/postprocessors/FindValueOnLine.C

```diff
@@ -0,0 +1,121 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "FindValueOnLine.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<FindValueOnLine>()
+{
+  InputParameters params = validParams<GeneralPostprocessor>();
+  params.addClassDescription("Find a specific target value along a sampling line. The variable values along the line should change monotonically. The target value is searched using a bisection algorithm.");
+  params.addParam<Point>("start_point", "Start point of the sampling line.");
+  params.addParam<Point>("end_point", "End point of the sampling line.");
+  params.addParam<Real>("target", "Target value to locate.");
+  params.addParam<unsigned int>("depth", 30, "Maximum number of bisections to perform.");
+  params.addParam<Real>("tol", 1e-10, "Stop search if a value within a symmetric interval of this with around the target is found.");
```

Comment:
The doco is confusing to me because i’m unsure whether “tol” relates to a line-segment length or whether it relates to an acceptable imprecision in the target value.

a


Ph: +61 7 3327 4497.  Fax: +61 7 3327 4666
Queensland Centre for Advanced Technologies
PO Box 883, Kenmore, Qld, 4069

From: Daniel Schwen [mailto:notifications@github.com]
Sent: Thursday, 5 May 2016 8:56 AM
To: idaholab/moose <moose@noreply.github.com>
Cc: Wilkins, Andy (Energy, Pullenvale) <Andrew.Wilkins@csiro.au>; Comment <comment@noreply.github.com>
Subject: Re: [idaholab/moose] Add find value by bisection (#6891)


In framework/src/postprocessors/FindValueOnLine.C<https://github.com/idaholab/moose/pull/6891#discussion_r62128545>:

> +/*             See LICENSE for full restrictions                */

> +/****************************************************************/

> +

> +#include "FindValueOnLine.h"

> +#include "MooseMesh.h"

> +

> +template<>

> +InputParameters validParams<FindValueOnLine>()

> +{

> +  InputParameters params = validParams<GeneralPostprocessor>();

> +  params.addClassDescription("Find a specific target value along a sampling line. The variable values along the line should change monotonically. The target value is searched using a bisection algorithm.");

> +  params.addParam<Point>("start_point", "Start point of the sampling line.");

> +  params.addParam<Point>("end_point", "End point of the sampling line.");

> +  params.addParam<Real>("target", "Target value to locate.");

> +  params.addParam<unsigned int>("depth", 30, "Maximum number of bisections to perform.");

> +  params.addParam<Real>("tol", 1e-10, "Stop search if a value within a symmetric interval of this with around the target is found.");

The width of the interval?

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/idaholab/moose/pull/6891/files/792f7c203de35dbecf55ee5292c1e8459e78d6cd#r62128545>